### PR TITLE
Day6

### PR DESCRIPTION
**Pull request by ryepup**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	os.Chdir(dir)
+
+	writeTemplate(
+		day,
+		fmt.Sprintf("day%v_test.go", day),
+		"test",
+		day)
+
+	for _, part := range []int{1, 2} {
```
Comment (ryepup) [inline main.go]: is this loop worth it?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		writeTemplate(
+			day,
+			fmt.Sprintf("part%v.go", part),
+			"part",
+			struct{ Day, Part int }{day, part})
+	}
+
+	err = os.WriteFile("test.txt", []byte("TODO: copy in test input\n"), 0644)
```
Comment (ryepup) [inline main.go]: probably want to prevent overwites here too

Comment (ryepup) [reply main.go]: done

File: main.go (hunk 1 above)
Comment (ryepup) [reply main.go]: not after a little simplification

File: day6/day6_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+func TestPart1(t *testing.T) {
+	testCases := map[string]int{"test.txt": 5934, "input.txt": 355386}
+	for path, expected := range testCases {
+		t.Run(path, func(t *testing.T) {
+			if expected == -1 {
+				t.Skip("TODO: provide an expected value")
+			}
+			n, err := Part1(path)
+			require.Nil(t, err)
```
Comment (UnwashedMeme) [inline day6/day6_test.go:18]: I have a soft spot for the functionally equivalent `require.NoError(t, err)`. The error message it generates is tailored to errors.

**Pull request merged**